packages/typespec-ts/static/static-helpers/urlTemplate.ts

```diff
@@ -0,0 +1,195 @@
+//---------------------
+// interfaces
+//---------------------
+interface ValueOptions {
+    isFirst: boolean; // is first value in the expression
+    op?: string; // operator
+    varValue?: any; // variable value
+    varName?: string; // variable name
+    modifier?: string; // modifier e.g *
+    reserved?: boolean; // if true we'll keep reserved words with not encoding
+}
+
+export interface UrlTemplateOptions {
+    // if set to true, reserved characters will not be encoded
+    allowReserved?: boolean;
+}
+
+// ---------------------
+// helpers
+// ---------------------
+function encodeValue(val: string, reserved?: boolean, op?: string) {
```

Comment:
There are many  common utility classes so please ensure we implemented those functions ourselves